src/App.js

```diff
@@ -6,32 +6,39 @@ import { useEffect, useState } from "react";
 import axios from "axios";
 
 const App = () => {
-  const [inputLanguage, setInputLanguage] = useState("English");
-  const [outputLanguage, setOutputLanguage] = useState("Japanese");
+  const [inputLanguage, setInputLanguage] = useState("English (en)");
+  const [outputLanguage, setOutputLanguage] = useState("Japanese (ja)");
   const [showModal, setShowModal] = useState(null);
   const [languages, setLanguages] = useState(null);
   const [textToTranslate, setTextToTranslate] = useState("");
   const [translatedText, setTranslatedText] = useState("");
 
-  const getLanguages = async() => {
+  const extractLanguageCode = (languageString) => {
+    const match = languageString.match(/\((\w+)\)/); // This regex matches the content within parentheses.
+    if (match && match[1]) {
+      return match[1]; // Returns the matched language code (in this case, "en").
+    }
+    return null; // Return null if no match is found.
+  }
+
+  const getLanguages = async () => {
     const response = await axios.get("http://localhost:8000/languages");
     setLanguages(response.data);
   }
 
   const translate = async () => {
     const data = {
-      textToTranslate, outputLanguage, inputLanguage
+      textToTranslate,
+      outputLanguage: extractLanguageCode(outputLanguage),
+      inputLanguage: extractLanguageCode(inputLanguage)
     }
     const response = await axios.get("http://localhost:8000/translation", {
       params: data,
     });
-    setTranslatedText(response.data);
+    // response.data returns in format {text: "", detectedSourceLang: ""}
+    setTranslatedText(response.data?.text);
```

Comment:
Also, this was updated because `response.data` returns `{text: "", detectedSourceLang: ""}`

Reply:
I got it. 
I checked your point, reading [the official document](https://www.npmjs.com/package/deepl-node#translating-text) one more time.